datahugger/services.py

```diff
@@ -425,3 +425,25 @@ class SeaNoeDataset(DatasetDownloader):
     ATTR_SIZE_JSONPATH = "size"
     ATTR_HASH_JSONPATH = "checksum"
     ATTR_HASH_TYPE_VALUE = "sha256"
+
+
+class B2shareDataset(DatasetDownloader):
+    """Downloader for B2Share repository."""
+
+    REGEXP_ID = r"b2share\.eudat\.eu\/records\/(?P<record_id>[0-9a-z]+)"
+
+    # the base entry point of the REST API
+    API_URL = "https://b2share.eudat.eu/api/"
+
+    # the files and metadata about the dataset
+    API_URL_META = "{api_url}records/{record_id}"
+    META_FILES_JSONPATH = "files[*]"
+
+    # paths to file attributes
+    ATTR_NAME_JSONPATH = "key"
+    ATTR_SIZE_JSONPATH = "size"
+    ATTR_HASH_JSONPATH = "checksum"
+    ATTR_HASH_TYPE_VALUE = "md5"
+
+    def _get_attr_link(self, record, base_url=None):
+        return f"{base_url}/files/{self._params['record_id']}/{record['key']}"
```

Comment:
```suggestion

    # paths to file attributes
    ATTR_NAME_JSONPATH = "key"
    ATTR_FILE_LINK_JSONPATH = "ePIC_PID"
    ATTR_SIZE_JSONPATH = "size"
    ATTR_HASH_JSONPATH = "checksum"
    ATTR_HASH_TYPE_VALUE = "md5"
```

This seems to work. I'm clueless about whats going on.